content/6_Advanced/LCT.mdx

```diff
@@ -30,18 +31,309 @@ frequency: 1
 		title="Splay Tree Paper"
 		url="https://www.cs.cmu.edu/~sleator/papers/self-adjusting.pdf"
 	/>
+	<Resource
+		source="CMU"
+		title="Splay Tree Proof"
+		url="https://github.com/Bibeknam/algorithmtutorprograms/blob/master/data-structures/splay-trees/SplayTree.cpp"
+	>
+		Uses less math than standard proof
+	</Resource>
 	<Resource
 		source="GH"
 		title="Splay Tree Implementation"
 		url="https://github.com/Bibeknam/algorithmtutorprograms/blob/master/data-structures/splay-trees/SplayTree.cpp"
 	/>
 </Resources>
 
-A splay tree is a type of self-balancing binary search tree that supports efficient
-implementation of operations such as finding an element, deleting an element,
-splitting a tree, and joining two trees.
+A splay tree is a type of self-balancing binary search tree that supports
+efficient implementation of operations such as finding an element, deleting an
+element, splitting a tree, and joining two trees.
+
+When a node in a splay tree is accessed, a splay operation is performed on the
+node that moves it to the root of the tree while also roughly balancing the
+tree. In one splay operation on $x$, the depths of the nodes on the path from
+the root to $x$ are roughly halved (as seen in the animations below).
+
+<video width="960" height="720" controls>
+	<source src="/animations/splay.mp4" type="video/mp4" />
+</video>
+
+This means that the high cost to access a very deep node is countered by the "work" it does to balance the tree. Unfortunately, this observation alone is not enough to prove the amortized
+$\mathcal{O}(\log n)$ complexity; the full proof remains quite involved.
+
+## Implicit Splays
+
+<FocusProblem problem="dsrars" />
+
+Much like [implicit treaps](https://usaco.guide/adv/treaps#implicit-treaps),
+implicit splays also exist.
+
+### Implementation
+
+**Time Complexity:** $\mathcal{O}((N + Q)\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <functional>
+#include <iostream>
+using ll = long long;
+using namespace std;
+
+// BeginCodeSnip{ModInt class}
+struct mint {
+	const static ll M = 998244353;
+	ll v = 0;
+	mint() {}
+	mint(ll v) { this->v = (v % M + M) % M; }
+	mint operator+(const mint &o) const { return v + o.v; }
+	mint &operator+=(const mint &o) {
+		v = (v + o.v) % M;
+		return *this;
+	}
+	mint operator*(const mint &o) const { return v * o.v; }
+	mint operator-(const mint &o) const { return v - o.v; }
+	mint &operator-=(const mint &o) {
+		mint t = v - o.v;
+		v = t.v;
+		return *this;
+	}
+	friend istream &operator>>(istream &s, mint &v) {
+		s >> v.v;
+		return s;
+	}
+	friend ostream &operator<<(ostream &s, const mint &v) {
+		s << v.v;
+		return s;
+	}
+};
+// EndCodeSnip
+
+struct Line {
+	mint w = 1, b = 0;
```

Comment:
uh what's w and b again

Reply:
w is slope, b is y-intercept (like y = wx + b, but i could change to mx + b if you want)

Reply:
ik you probably documented this alr but could you add a comment agai